projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts

```diff
@@ -83,7 +84,7 @@ export class IgxForOfContext<T> {
     standalone: true
 })
 // eslint-disable @angular-eslint/no-conflicting-lifecycle
-export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestroy, AfterViewInit {
+export class IgxForOfDirective<T, U extends Array<T> = Array<T>> implements OnInit, OnChanges, DoCheck, OnDestroy, AfterViewInit {
```

Comment:
Unfortunately does not work with `Array<T>`, but works with `NgIterable<T>`. 
We do depend on the array's api a lot and changing it to use iterable would be a bit of a breaking change and a bit of a pain. Maybe we could use something like:
```
export declare type IgIterable<T> = Array<T> & Iterable<T>;
```

Reply:
@MayaKirova Can you check how it is now. For some reason when using the `Array<T>` type it does not recognize it, but using tuple `T[]` seems to do the trick. I guess because tuple recognizes the types the array contains while the other type does not.

Reply:
Looks ok with `T[]`